frontend/app/routes/protected/person-case/birth-details.tsx

```diff
@@ -0,0 +1,275 @@
+import type { ChangeEvent } from 'react';
+import { useId, useState } from 'react';
+
+import { data, useFetcher } from 'react-router';
+import type { RouteHandle } from 'react-router';
+
+import { faExclamationCircle, faXmark } from '@fortawesome/free-solid-svg-icons';
+import type { SessionData } from 'express-session';
+import { useTranslation } from 'react-i18next';
+import * as v from 'valibot';
+
+import type { Route, Info } from './+types/birth-details';
+
+import { requireAuth } from '~/.server/utils/auth-utils';
+import { i18nRedirect } from '~/.server/utils/route-utils';
+import { Button } from '~/components/button';
+import { FetcherErrorSummary } from '~/components/error-summary';
+import { InputField } from '~/components/input-field';
+import type { InputRadiosProps } from '~/components/input-radios';
+import { InputRadios } from '~/components/input-radios';
+import { InputSelect } from '~/components/input-select';
+import { PageTitle } from '~/components/page-title';
+import { Progress } from '~/components/progress';
+import { AppError } from '~/errors/app-error';
+import { ErrorCodes } from '~/errors/error-codes';
+import { getFixedT } from '~/i18n-config.server';
+import { handle as parentHandle } from '~/routes/protected/layout';
+import { getLanguage } from '~/utils/i18n-utils';
+import { REGEX_PATTERNS } from '~/utils/regex-utils';
+import { trimToUndefined } from '~/utils/string-utils';
+
+type BirthDetailsSessionData = NonNullable<SessionData['inPersonSINCase']['birthDetails']>;
+
+const REQUIRE_OPTIONS = { yes: 'Yes', no: 'No' } as const;
+const COUNTRIES: string[] = ['CAN', 'FRA'];
+
+export const handle = {
+  i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
+} as const satisfies RouteHandle;
+
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+  const currentNameInfo = context.session.inPersonSINCase?.currentNameInfo;
+  const preferredSameAsDocumentName = currentNameInfo?.preferredSameAsDocumentName;
+  return {
```

Comment:
I find the code difficult to read when there's no vertical whitespace at all. It's hard to know where one block of code ends and the next block begins.

```suggestion
  requireAuth(context.session, new URL(request.url), ['user']);

  const t = await getFixedT(request, handle.i18nNamespace);
  const currentNameInfo = context.session.inPersonSINCase?.currentNameInfo;
  const preferredSameAsDocumentName = currentNameInfo?.preferredSameAsDocumentName;

  return {
```